src/core/server/http_resources/get_apm_config.ts

```diff
@@ -6,36 +6,41 @@
  * Side Public License, v 1.
  */
 
-import { getConfig } from '../../../apm';
 import agent from 'elastic-apm-node';
+// @ts-expect-error apm module is a js file outside of core (need to split APM/rum configuration)
+import { getConfig } from '../../../apm';
 
-const apmEnabled = getConfig()?.active;
+let apmActive: boolean;
+const isApmActive = () => {
+  if (apmActive === undefined) {
+    apmActive = getConfig()?.active;
+  }
+  return apmActive;
+};
 
-export function getApmConfig(requestPath) {
-  if (!apmEnabled) {
+export const getApmConfig = (requestPath: string) => {
+  if (!isApmActive()) {
     return null;
   }
-  const config = {
+
+  const config: Record<string, any> = {
     ...getConfig('kibana-frontend'),
     pageLoadTransactionName: requestPath,
   };
 
-  /**
-   * Get current active backend transaction to make distrubuted tracing
-   * work for rendering the app
-   */
+  // Get current active backend transaction to make distributed tracing
+  // work for rendering the app
   const backendTransaction = agent.currentTransaction;
 
   if (backendTransaction) {
-    const { sampled, traceId } = backendTransaction;
+    const { sampled, traceId } = backendTransaction as any;
```

Comment:
`agent.currentTransation` is an `apm.Transaction`, but the type does not declare a `sampled` or `traceId` field, so I was forced to forcecast to `any` here. cc @elastic/apm-agent-node-js